lib/src/domain/usecase/implementation/get_now_playing_usecase.dart

```diff
@@ -0,0 +1,14 @@
+import '../../../data/repositories/movie_repository.dart';
+import '../../entity/response.dart';
+import '../i_usecase.dart';
+
+class GetNowPlayingUseCase implements IUseCase {
+  final MovieRepository repository;
+
+  const GetNowPlayingUseCase({this.repository = const MovieRepository()});
```

Comment:
You have 4 UseCases, all uses a repository, the same repository, but you are creating 4 new instances, try to change this to create only one instance of the repository and shared it to all the use cases

Reply:
done